src/main/java/org/folio/service/export/LocalFileSystemExportService.java

```diff
@@ -1,62 +1,114 @@
 import org.marc4j.MarcJsonReader;
 import org.marc4j.MarcReader;
 import org.marc4j.MarcStreamWriter;
 import org.marc4j.MarcWriter;
 import org.marc4j.marc.Record;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.lang.invoke.MethodHandles;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.List;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
 
 import static org.apache.commons.lang3.ArrayUtils.isNotEmpty;
 import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.folio.util.ErrorCode.ERROR_MARC_RECORD_CANNOT_BE_CONVERTED;
 
 @Service
 public class LocalFileSystemExportService implements ExportService {
   private static final Logger LOGGER = LogManager.getLogger(MethodHandles.lookup().lookupClass());
 
+  private static final String GENERAL_INFO_FIELD_TAG_NUMBER = "999";
+  private static final String INDICATOR_VALUE = "f";
+  private static final String FIRST_INDICATOR = "ind1";
+  private static final String SECOND_INDICATOR = "ind2";
+  private static final String INSTANCE_ID_KEY = "i";
+  private static final String SUBFIELDS_KEY = "subfields";
+  private static final int SINGLE_INSTANCE_INDEX = 0;
+  private static final String INSTANCES = "instances";
+  private static final int SINGLE_INSTANCE = 1;
+
   @Autowired
   @Qualifier("LocalFileSystemStorage")
   private FileStorage fileStorage;
   @Autowired
   private ExportStorageService exportStorageService;
   @Autowired
   private ErrorLogService errorLogService;
+  @Autowired
+  private InventoryClient inventoryClient;
 
   @Override
-  public void exportSrsRecord(List<String> jsonRecords, FileDefinition fileDefinition) {
+  public void exportSrsRecord(List<String> jsonRecords, ExportPayload exportPayload) {
+    FileDefinition fileDefinition = exportPayload.getFileExportDefinition();
+    String jobExecutionId = exportPayload.getJobExecutionId();
+    OkapiConnectionParams params = exportPayload.getOkapiConnectionParams();
     if (CollectionUtils.isNotEmpty(jsonRecords) && fileDefinition != null) {
       for (String jsonRecord : jsonRecords) {
-        byte[] bytes = convertJsonRecordToMarcRecord(jsonRecord);
         try {
+          byte[] bytes = convertJsonRecordToMarcRecord(jsonRecord);
           if (isNotEmpty(bytes)) {
             fileStorage.saveFileDataBlocking(bytes, fileDefinition);
           }
+        } catch (MarcException e) {
+          String instId = getInstanceIdFromMarcRecord(new JsonObject(jsonRecord));
+          inventoryClient.getInstancesByIds(Collections.singletonList(instId), jobExecutionId, params, SINGLE_INSTANCE).ifPresent(instancesJson -> {
```

Comment:
Here the blocking client is used since all the flow is performed on a separate thread pool and doesn't block the main event loop thread.